spec/bson/byte_buffer_write_spec.rb

```diff
@@ -585,6 +585,15 @@
     end
   end
 
+  describe '#put_uint32' do
+    context 'when argument is a float' do
+      it 'appends the int32 to the byte buffer' do
+        expect{ buffer.put_uint32(4.934) }.to raise_error(ArgumentError, "put_uint32; incorrect type: float, expected: integer")
+      end
+
+    end
+  end
```

Comment:
I'd love to see some examples that test the extreme cases that should work or raise an error -- i.e. 0 and 4294967295, and -1 and 4294967296.

Reply:
I think I put those in byte_buffer_spec.rb, would you like me to transfer them?

Reply:
Yeah I think so -- because this method is implemented in the `write` file of the C extension, I think it makes sense to do the bulk of the testing in this file. Thanks for asking :)

Reply:
done!